libs/s25main/controls/ctrlPreviewMinimap.cpp

```diff
@@ -80,8 +82,8 @@ void ctrlPreviewMinimap::SetMap(const glArchivItem_Map* const s2map)
             // Startposition eines Spielers an dieser Stelle?
             if(s2map->GetMapDataAt(MAP_TYPE, x, y) != 0x80)
                 continue;
-            unsigned player = s2map->GetMapDataAt(MAP_LANDSCAPE, x, y);
-            if(player < MAX_PLAYERS)
+            const unsigned player = s2map->GetMapDataAt(MAP_LANDSCAPE, x, y);
+            if(player < players.size())
```

Comment:
I thought the numbers must not be continuous on some maps?

Reply:
I'd be surprised by that. The editor only allows you to set them contiguous. But it doesn't matter for this code. This simply places a colored square at each position where a player HQ is set. I only changed the `MAX_PLAYERS` to `players.size()` where `players` is a `MAX_PLAYERS` sized array, so no change here. I only wanted to avoid the duplication and make it clear no OOB access is happening

BUT: Maybe we should validate that all players are set. For that I'd move the `Point` class to libutil and let an s2map in libsiedler2 return an array with player positions for each player in the map. This allows to validate in libsiedler2 that each player has a unique start position and in s25client that there are as many start positions as in the header and each is set. I'd do the latter in s25client because it might be that we want players without a start position. See related issue #935 where a mission seems to have that case